### PR TITLE
Add "Frontend Performance Optimization" article

### PR DESCRIPTION
**Pull request by bonnie57**
Add "Frontend Performance Optimization" article

### CODE REVIEW

**Comment by live9080** [comment]
Can you please rebase or merge the latest 'dev' branch? We have added an action in the 'dev' branch that automatically generates a preview URL. After you update your PR, you should be able to get the preview URL added by the bot(like the screenshot below) so that reviewers can access the preview.

image

**Comment by live9080** [comment]
Could you please rebase or merge the latest dev branch? We've added an action in the dev branch to automatically generate a preview URL. After you update the PR, you'll get a preview URL added by the bot (just like the screenshot below) so that reviewers can take a look.

image

**Review by shan-57blocks — APPROVED** [review]

**Comment by shan-57blocks** [comment]
@live9080 Approved

**Comment by Martha-luo** [comment]
Screenshot 2024-05-29 at 15 46 23

File: articles/Frontend Performance Optimization/README.md [hunk 1]
```diff
@@ -0,0 +1,519 @@
...
+];
+```
+
+### Http cache
+
+Usually, we configure appropriate directives based on the update frequency of resources to retrieve them from the cache. This helps reduce request frequency and improve loading efficiency. This involves specific configurations for responses and requests. The specific configuration directives may vary depending on the circumstances. For detailed information, please click [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control#cache_directives).
+
+1. For HTML files with a high update frequency, the directive settings commonly used are:
+
+```ts
```
Comment (live9080) [inline articles/Frontend Performance Optimization/README.md:203]: If the code is in a list, it is best to indent the code to ensure that the code is aligned with the list item.

This is also the case elsewhere in the article.

File: articles/Frontend Performance Optimization/README.md [hunk 2]
```diff
@@ -0,0 +1,519 @@
...
+<img src="flower-large.jpg" srcset="example-small.jpg 480w, example-large.jpg 1080w" sizes="50vw">
+```
+
+**`480w`** indicates informing the browser, in situations where downloading the image is not necessary, that the width is equivalent to 480 pixels.
+
+**`sizes`** specifies the expected display size of the image.
+
+- **width 和 height：**
```
Comment (live9080) [inline articles/Frontend Performance Optimization/README.md:345]: en.

File: articles/Frontend Performance Optimization/README.md [hunk 3]
```diff
@@ -0,0 +1,519 @@
...
+const observer = new PerformanceObserver((list) => {
+  list.getEntries().forEach((entry) => {
+    console.log(entry);
+  });
+});
+observer.observe({ type: "paint", buffered: true });
+```
+
+![FCP](FCP-object.png)
+In the given metric:
```
Comment (live9080) [inline articles/Frontend Performance Optimization/README.md:68]: Add the corresponding sections before and after the picture

**Comment by Martha-luo** [comment]
1. 建议对文章的章节再review一下，是否每个二级三级标题归属于正确的上级标题。
2. 文章里面的大小写不统一，三级标题有些大写有些小写，还有中文字符，请核对。
 
以下是一些截图请参考：
Screenshot 2024-07-02 at 22 19 26
Screenshot 2024-07-02 at 22 55 34
Screenshot 2024-07-02 at 22 54 00
Screenshot 2024-07-02 at 22 52 49
Screenshot 2024-07-02 at 22 50 45
Screenshot 2024-07-02 at 22 48 02
Screenshot 2024-07-02 at 22 43 49
Screenshot 2024-07-02 at 22 41 43
Screenshot 2024-07-02 at 22 38 15
Screenshot 2024-07-02 at 22 35 27
Screenshot 2024-07-02 at 22 33 32
Screenshot 2024-07-02 at 22 32 43
Screenshot 2024-07-02 at 22 30 39
Screenshot 2024-07-02 at 22 26 20

**Comment by Martha-luo** [comment]
Screenshot 2024-07-03 at 14 19 06
Screenshot 2024-07-03 at 14 17 22
Screenshot 2024-07-03 at 14 15 20

**Review by live9080 — APPROVED** [review]

**Pull request merged**